cluster_ctl

```diff
@@ -410,48 +421,80 @@ def generate_config_files():
       )
     shutil.copytree("dashboards", "config_files/grafana/dashboards",
                     dirs_exist_ok=True)
-    # nginx config for grafana and locust. This implements a simple cookie-based
-    # authentication. An unauthenticated user receives 403 for all requests
-    # except /secret/path, which will set the authentication cookie and redirect
-    # to /path.
-    gui_secret = get_gui_secret()
-    for config_file_name, service_port in [("nginx_grafana.conf", 3000),
-                                           ("nginx_locust.conf", 8089)]:
-        w(f"config_files/{config_file_name}",
-          f'error_log {RUN_DIR}/nginx/nginx_error.log error;',
-          f'pid {RUN_DIR}/nginx/nginx.pid;',
-          'events {',
-          '  worker_connections 4096;',
-          '}',
-          'http {',
-          '  map $http_upgrade $connection_upgrade {',
-          '    default upgrade;',
-          '    ""      close;',
-          '  }',
-          f'  client_body_temp_path {RUN_DIR}/nginx/client_temp;',
-          f'  proxy_temp_path       {RUN_DIR}/nginx/proxy_temp;',
-          f'  fastcgi_temp_path     {RUN_DIR}/nginx/fastcgi_temp;',
-          f'  uwsgi_temp_path       {RUN_DIR}/nginx/uwsgi_temp;',
-          f'  scgi_temp_path        {RUN_DIR}/nginx/scgi_temp;',
-          '  server {',
-          f'    access_log {RUN_DIR}/nginx/nginx_access.log;',
-          '    listen 8080;',
-          f'    location ~ ^/{gui_secret}(/.*)$ {{',
-          f'      add_header Set-Cookie "X-AUTH={gui_secret}; Path=/";',
-          '      return 302 $1;',
-          '    }',
-          '    location / {',
-          f'      if ($http_cookie !~* "X-AUTH={gui_secret}") {{',
-          '        return 403;',
-          '      }',
-          '      proxy_http_version 1.1;',
-          '      proxy_set_header Upgrade $http_upgrade;',
-          '      proxy_set_header Connection $connection_upgrade;',
-          f'      proxy_pass http://localhost:{service_port};',
-          '    }',
-          '  }',
-          '}',
-          )
+    # nginx config to expose grafana, locust and demo UI on the first bench
+    # node. This implements a simple cookie-based authentication. An
+    # unauthenticated user receives 403 for all requests except
+    # /secret=SECRET/path, which will set the authentication cookie and redirect
+    # to /path. location / is proxied to the demo UI without authentication, on
+    # purpose.
+    w("config_files/nginx.conf",
+      f'error_log {RUN_DIR}/nginx/nginx_error.log error;',
+      f'pid {RUN_DIR}/nginx/nginx.pid;',
+       'events {',
+       '  worker_connections 4096;',
```

Comment:
The number of worker_connections is larger than the default max open files (= 1024), so would
recommend setting this to much lower value.

Reply:
@mronstro There's also a conversation at https://github.com/logicalclocks/rondb-tools/commit/72699f5148865faabea691b0372136ba50c6dc47#r162533637